### PR TITLE
Update notice for default cart and checkout

### LINKED ISSUE #11777: Update notice for default cart and checkout
## Is your feature request related to a problem? Please describe.

When a merchant adds the Cart or Checkout block to a page, but does not define the corresponding pages as the cart or checkout page via `/wp-admin/admin.php?page=wc-settings&tab=advanced`, the following notices are visible in the sidebar:

<table>
<tr>
<td valign="top">Current notice for Cart block:
<br><br>
Screenshot 2023-11-15 at 10 42 25
</td>
<td valign="top">Current notice for Checkout block:
<br><br>
Screenshot 2023-11-15 at 10 41 46
</td>
</tr>
</table>

## Describe the solution you'd like

In p1699988291133359/1699592175.220079-slack-C02UBB1EPEF, @elizaan36 asked to replace the button in the `<DefaultNotice>` component with an inline link and to display an info notice instead of a warning notice.

<table>
<tr>
<td valign="top">Expected notice for Cart block:
<br><br>
Screenshot 2023-11-15 at 10 54 06
</td>
<td valign="top">Expected notice for Checkout block:
<br><br>
Screenshot 2023-11-15 at 10 55 22
</td>
</tr>
</table>

## Describe alternatives you've considered

n/a

## Additional context

n/a

### PR DESCRIPTION
**Pull request by nielslange**
## What

Fixes #11777

## Why

In p1699988291133359/1699592175.220079-slack-C02UBB1EPEF, @elizaan36 asked to replace the button in the <DefaultNotice> component with an inline link and to display an info notice instead of a warning notice.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Create a test page and add the Cart block to it.
2. Open the settings sidebar.
3. Select the Cart block.
4. Verify that the notice _"If you would like to use this block as your default cart, [update your page settings](#)."_ no longer appears as a warning (yellow border and background), but as info (blue border, white background).
5. Click the link _"[update your page settings](#)"_.
6. Verify that the notice _"Page settings updated"_ becomes visible.
7. Repeat steps 1. until 6. with the Checkout block. In step 4., the message should say _"... as your default checkout, ..."_ instead of _"... as your default cart, ..."_.

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

### Cart block

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2023-11-21 at 18 17 37
</td>
<td valign="top">After:
<br><br>
Screenshot 2023-11-21 at 17 57 27
</td>
</tr>
</table>

### Checkout block

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2023-11-21 at 18 11 48
</td>
<td valign="top">After:
<br><br>
Screenshot 2023-11-21 at 18 08 52
</td>
</tr>
</table>

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:

* [x] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Update notice for default cart and checkout

### CODE REVIEW

File: assets/js/editor-components/default-notice/index.tsx [hunk 1]
```diff
@@ -112,6 +98,30 @@ export function DefaultNotice( { block }: { block: string } ) {
 		saveEntityRecord,
 		slug,
 	] );
+	const noticeContent = createInterpolateElement(
+		__(
+			'If you would like to use this block as your default <blockTypeLink/>, <updateLink>update your page settings</updateLink>.',
+			'woo-gutenberg-products-block'
+		),
```
Comment (senadir) [inline assets/js/editor-components/default-notice/index.tsx]: Place holding a translatable string with another translatable string is discouraged and I don't think it should be done, as string change meaning on context and the translator should have the liberty to translate a full thing, not a fragmented one.

So I'd say it's fine to dubplicate this.

File: assets/js/editor-components/default-notice/index.tsx [hunk 2]
```diff
@@ -112,6 +98,30 @@ export function DefaultNotice( { block }: { block: string } ) {
...
+			blockTypeLink: (
+				<span>
+					{ block === 'checkout'
+						? __( 'checkout', 'woo-gutenberg-products-block' )
+						: __( 'cart', 'woo-gutenberg-products-block' ) }
+				</span>
+			),
+			updateLink: (
+				// eslint-disable-next-line jsx-a11y/anchor-is-valid
+				<a href="#" onClick={ updatePage }>
```
Comment (senadir) [inline assets/js/editor-components/default-notice/index.tsx]: Clicking this multiple times will trigger the call several times, but that doesn't seem to cause a problem in my testing.

File: assets/js/editor-components/default-notice/index.tsx [hunk 3]
```diff
@@ -112,6 +98,30 @@ export function DefaultNotice( { block }: { block: string } ) {
...
+		{
+			blockTypeLink: (
+				<span>
+					{ block === 'checkout'
+						? __( 'checkout', 'woo-gutenberg-products-block' )
+						: __( 'cart', 'woo-gutenberg-products-block' ) }
+				</span>
+			),
+			updateLink: (
+				// eslint-disable-next-line jsx-a11y/anchor-is-valid
```
Comment (senadir) [inline assets/js/editor-components/default-notice/index.tsx]: It should be noted that initially, this was a link, then we switched to a button due to this problem, we're back again at it being a link. I think it should be fine.

**Review by senadir — APPROVED** [review]
I'm preapproving but I'd love to see the translation issue fixed, as the i18n also pointed out this is discouraged behavior.

File: assets/js/editor-components/default-notice/index.tsx (hunk 1 above)
Comment (nielslange) [reply assets/js/editor-components/default-notice/index.tsx]: I practice DRY, I decided not to duplicate the code. That said, I see your point and I agree that it's easier for translators if they have the full context. I've just adjusted this.

File: assets/js/editor-components/default-notice/index.tsx (hunk 3 above)
Comment (nielslange) [reply assets/js/editor-components/default-notice/index.tsx]: Thanks for pointing this out, @senadir. I tried to accomplish the request using a button, but when doing so, the link text was displayed below the regular text and not inline. The only way to get the link inline, was by using [`display: contents`](https://caniuse.com/?search=contents), which is only fully supported by Safari on iOS. That's why I decided to switch back to an anchor element, to display the link as requested in #11777.

**Comment by nielslange** [comment]
> I'm preapproving but I'd love to see the translation issue fixed, as the i18n also pointed out this is discouraged behavior.

Thanks for your review, @senadir, and for pointing out the i18n issue. I've addressed it, as suggested by you.
